en/docs/install-and-setup/upgrading-wso2-api-manager/260-to-410/upgrading-from-260-to-410.md

```diff
@@ -0,0 +1,1282 @@
+    
+    !!! note
+        Note that if you want to use the latest user store, update the <API-M_4.1.0_HOME>/repository/conf/deployment.toml as follows after the identity migration,
+
+        ```
+        [user_store]
+        type = "database_unique_id"
+        ``` 
+
+    !!! warning "Troubleshooting"
+        When running the above step if you encounter the following error message, follow the steps in this section. Note that this error could occur only if the identity tables contain a huge volume of data.
+
+        Sample exception stack trace is given below.
+        ```
+        ERROR {org.wso2.carbon.registry.core.dataaccess.TransactionManager} -  Failed to start new registry transaction. {org.wso2.carbon.registry.core.dataaccess.TransactionManager} org.apache.tomcat.jdbc.pool.PoolExhaustedException: [pool-30-thread-11] Timeout: Pool empty. Unable to fetch a connection in 60 seconds, none available[size:50; busy:50; idle:0; lastwait:60000
+        ```
+
+        1.  Add the following property in `<API-M_HOME>/repository/conf/deployment.toml` to a higher value (e.g., 10)
+            ```
+            [indexing]
+            frequency= 10
+            ```
+
+        2.  Re-run the command above.
+
+        **Make sure to revert the change done in Step 1 , after the migration is complete.**
+
+10.  After you have successfully completed the migration, stop the server and remove the following files and folders.
+
+    -   Remove the `org.wso2.carbon.is.migration-x.x.x.jar` file, which is in the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+    -   Remove the `migration-resources` directory, which is in the `<API-M_4.1.0_HOME>` directory.
+
+    -   If you ran WSO2 API-M as a Windows Service when doing the identity component migration , then you need to remove the following parameters in the command line arguments section (CMD_LINE_ARGS) of the api-manager.bat file.
+
+        ```
+        -Dmigrate -Dcomponent=identity
+        ```
+
+### Step 4: Migrate the API Manager Components
+
+You have to run the following migration client to update the API Manager artifacts.
+
+1.  Navigate to the [latest release tag (v4.1.0.x)](https://github.com/wso2-extensions/apim-migration-resources/releases) under **Assets** and download the `wso2am-migration-4.1.0.x.zip`. Unzip it to a local directory.
+
+    !!! note
+
+        x of wso2am-migration-4.1.0.x.zip denotes the version number of the most recently-released migration resources.
+
+        The directory where the wso2am-migration-4.1.0.x.zip is unziped will be referred to as `<AM_MIGRATION_CLIENT_HOME>`
+
+2. Copy the `<AM_MIGRATION_CLIENT_HOME>/migration-resources` to the `<API-M_4.1.0_HOME>` directory.
+
+3. Copy the org.wso2.carbon.am.migration-4.1.0.x.jar file in the `<AM_MIGRATION_CLIENT_HOME>/dropins` directory into the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
```

Comment:
Add pre migration step to here

Reply:
fixed in https://github.com/wso2/docs-apim/pull/5985/commits/fa4963842f4845496ea960bd3e91b0977dd9d017